### DISCUSSION TITLE
classes/class_fastnoiselite

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#233] [User-contributed notes]
# classes/class_fastnoiselite

Inherits: Noise< Resource< RefCounted< Object Generates noise using the FastNoiseLite library. Description: This class generates noise using the FastNoiseLite library, which is a collection of seve...

https://docs.godotengine.org/en/stable/classes/class_fastnoiselite.html

**Comment by jokoon** [comment]
Can't fastnoiselite combine noise layers?

**Reply by Calinou** [reply]
Not directly, but you can read multiple FastNoiseLite instances and combine them manually in your code. This is also possible with NoiseTexture2D using the Image class' blending functions.

**Comment by AbidHasan95** [comment]
How to create a repeating pattern with a specified period? It can be used for creating a seamless endless map

**Reply by Pthread-Create** [reply]
if you want to do a repeating pattern in 2D like repeating in the X coordinate you can use a noise3D and get the value for a cylinder, if you want to repeat in the X and Y you can do a sphere in 3D etc etc

**Reply by psychicteeth** [reply]
Cylinder idea is good, but doesn't work if you need a 4th dimension (time). The only way then is to use the seamless image function combined with the offset property to generate the entire noise field at once.

**Comment by bsuodo** [comment]
I't doesn't seem to work all i get is an error saying: Line 3:Function "FastNoiseLite()" not found in base self.
I have no idea as to why i't shouldn't work.

**Reply by Calinou** [reply]
You should use `FastNoiseLite.new()`, not `FastNoiseLite()`.

**Reply by bsuodo** [reply]
Even with "FastNoiseLite.new()" it still doesn't work.

**Reply by Calinou** [reply]
Which Godot version are you using? FastNoiseLite was added in Godot 4.0, so it's not in 3.x. In Godot 3, use `OpenSimplexNoise.new()` instead (but keep in mind the set of methods it exposes is different from FastNoiseLite).

**Reply by bsuodo** [reply]
I’m currently using version 4.4.1. The latest stable version. Which means that FastNoiseLite can be used, but just doesn’t work.

**Comment by NailMinnigalin** [comment]
This article lacks a description of the methods.

**Reply by Calinou** [reply]
All properties and enums are documented in this class (there are no missing descriptions). Which property/enum are you referring to?

PS: It's preferable to [open an issue](https://github.com/godotengine/godot-docs) to request improvements to the documentation, so that it can be tracked more easily.
